src/PayplugWoocommerceHelper.php

```diff
@@ -436,17 +437,29 @@ public static function save_transaction_metadata( $order, $metadata ) {
 	 *
 	 * @return array
 	 */
-	public static function get_account_data_from_options() {
-		$options          = get_option( 'woocommerce_payplug_settings', [] );
-		$payplug_test_key = ! empty( $options['payplug_test_key'] ) ? $options['payplug_test_key'] : '';
-		$payplug_live_key = ! empty( $options['payplug_live_key'] ) ? $options['payplug_live_key'] : '';
-		if(empty($payplug_test_key) && empty($payplug_live_key)) {
-			return array();
-		}
-		$account = Authentication::getAccount(new Payplug($options['mode'] === 'yes' ? $payplug_live_key : $payplug_test_key));
-		$account['oney'] = $options['oney'];
-		$account['oneycgv'] = $options['oneycgv'];
-		return $account;
+	public static function get_account_data_from_options($force_update = false) {
+        $options          = get_option('woocommerce_payplug_settings', []);
+        $payplug_test_key = !empty($options['payplug_test_key']) ? $options['payplug_test_key'] : '';
+        $payplug_live_key = !empty($options['payplug_live_key']) ? $options['payplug_live_key'] : '';
+        if (empty($payplug_test_key) && empty($payplug_live_key)) {
+            return array();
+        }
+
+        $transient_key = PayplugGatewayOney3x::OPTION_NAME . ($options['mode'] === 'yes' ? "_live" : "_test");
+        $payplug_oney_config = get_transient($transient_key);
+        if ($payplug_oney_config && !$force_update) {
+            $account = $payplug_oney_config;
+        } else {
+            try {
+                $account = Authentication::getAccount(new Payplug($options['mode'] === 'yes' ? $payplug_live_key : $payplug_test_key));
```

Comment:
```suggestion
                $parameters_account = Authentication::getAccount(new Payplug($options['mode'] === 'yes' ? $payplug_live_key : $payplug_test_key));
```
Changement de variable suite à la suggestion prochaine qui évite un warning php car le format entre getAccount et get_transient n'est pas exactement le même

Reply:
Ok